23-input-scanning/exercises/02-unique-words/main.go

```diff
@@ -32,4 +38,24 @@ package main
 // ---------------------------------------------------------
 
 func main() {
+	var (
+		in          = bufio.NewScanner(os.Stdin)
+		totalUnique int
+		total       int
+		unique      = map[string]bool{}
+	)
+
+	in.Split(bufio.ScanWords)
+	for in.Scan() {
+		word := in.Text()
+		_, ok := unique[word]
+		if ok {
+			totalUnique--
```

Comment:
Over here you are first decrementing in some condition and then incrementing. So it is little confusing to understand.
Keep it simplified like if the string is not present then only it is unique, so inc it only then. This way, no need to first dec and then inc

Reply:
fixed